pkg/cli/cmd/pipelinerun/list.go

```diff
@@ -0,0 +1,129 @@
+package pipelinerun
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"text/tabwriter"
+	"text/template"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/formatted"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/results/pkg/cli/flags"
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+)
+
+const listTemplate = `{{- $size := len .PipelineRuns -}}{{- if eq $size 0 -}}
+No PipelineRuns found
+{{ else -}}
+NAMESPACE	UID	STARTED	DURATION	STATUS
+{{- range $_, $pr := .PipelineRuns }}
+{{ $pr.ObjectMeta.Namespace }}	{{ $pr.ObjectMeta.Name }}	{{ formatAge $pr.Status.StartTime $.Time }}	{{ formatDuration $pr.Status.StartTime $pr.Status.CompletionTime }}	{{ formatCondition $pr.Status.Conditions }}
+{{- end -}}
+{{- end -}}`
+
+type listOptions struct {
+	Namespace string
+	Limit     int
+}
+
+// listCommand initializes a cobra command to list PipelineRuns
+func listCommand(params *flags.Params) *cobra.Command {
+	opts := &listOptions{Limit: 0, Namespace: "default"}
+
+	eg := `List all PipelineRuns in a namespace 'foo':
+    tkn-results pipelinerun list -n foo
+
+List all PipelineRuns in 'default' namespace:
+    tkn-results pipelinerun list -n default
+`
+	cmd := &cobra.Command{
+		Use:     "list",
+		Aliases: []string{"ls"},
+		Short:   "List PipelineRuns in a namespace",
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Example: eg,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if opts.Limit < 0 {
+				return fmt.Errorf("limit was %d, but must be greater than 0", opts.Limit)
+			}
+
+			resp, err := params.ResultsClient.ListRecords(cmd.Context(), &pb.ListRecordsRequest{
+				Parent:   fmt.Sprintf("%s/results/-", opts.Namespace),
+				PageSize: int32(opts.Limit),
+				Filter:   `data_type==PIPELINE_RUN`,
+			})
+			if err != nil {
+				return fmt.Errorf("failed to list PipelineRuns from namespace %s: %v", opts.Namespace, err)
+			}
+			return printFormatted(cmd.OutOrStdout(), resp.Records, params.Clock)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Namespace, "namespace", "n", "default", "Namespace to list PipelineRuns in")
+	cmd.Flags().IntVarP(&opts.Limit, "limit", "l", 0, "Limit the number of PipelineRuns to return")
+	return cmd
+}
+
+func pipelineRunFromRecord(record *pb.Record) (*pipelinev1.PipelineRun, error) {
+	if record.Data == nil {
+		return nil, fmt.Errorf("record data is nil")
+	}
+	pr := &pipelinev1.PipelineRun{}
+	switch record.Data.GetType() {
+	case "tekton.dev/v1beta1.PipelineRun":
+		prV1beta1 := &pipelinev1beta1.PipelineRun{}
```

Comment:
You can add  //nolint:staticcheck adjacent to this line. It will fix the build test.

Reply:
Fixed and rebased.